### PR TITLE
Fixed useItem

### PR DESCRIPTION
**Pull request by KaffinPX**
useItem.js was broken i fixed it for multiple versions.

### CODE REVIEW

File: src/lib/plugins/useItem.js [hunk 1]
```diff
@@ -27,10 +40,24 @@ module.exports.server = (serv, { version }) => {
...
+        })
+      } else {
+        if (version === '1.8.8') {
+          serv.onItemPlace('spawn_egg', ({ item, player, placedPosition }) => {
+            serv.spawnMob(item.metadata, player.world, placedPosition)
+            return { id: -1, data: 0 }
+          })
+        } else {
+          serv.onItemPlace('spawn_egg', ({ item, player, placedPosition }) => {
+            serv.spawnMob(getEntID(item.nbt.value.EntityTag.value.id.value), player.world, placedPosition)
```
Comment (Karang) [inline src/lib/plugins/useItem.js:56]: You could have only 1 getEntID function if you call it like:
```js
serv.spawnMob(getEntID('minecraft:' + item.nbt.value.EntityTag.value.id.value), player.world, placedPosition)

```

File: src/lib/plugins/useItem.js [hunk 2]
```diff
@@ -27,10 +40,24 @@ module.exports.server = (serv, { version }) => {
...
-        serv.spawnMob(getEntID(heldItem.nbt.value.EntityTag.value.id.value), player.world, placedPosition)
-        return { id: -1, data: 0 }
-      })
+      if (version === '1.12.2') {
+        serv.onItemPlace('spawn_egg', ({ item, player, placedPosition }) => {
+          serv.spawnMob(getEntIDmc(item.nbt.value.EntityTag.value.id.value), player.world, placedPosition)
+          return { id: -1, data: 0 }
+        })
+      } else {
+        if (version === '1.8.8') {
```
Comment (Karang) [inline src/lib/plugins/useItem.js:49]: To check the version you need to use the supportFeature system (it must also work for other 1.8 versions (snapshots etc..)

Basically, you create an entry in https://github.com/PrismarineJS/flying-squid/blob/master/src/lib/features.json#L63
And check with
```js
if (serv.supportFeature('entityMCPrefixed')) {
```

File: src/lib/plugins/useItem.js (hunk 1 above)
Comment (KaffinPX) [reply src/lib/plugins/useItem.js:56]: Optimizing code wait a minute

File: src/lib/plugins/useItem.js (hunk 2 above)
Comment (KaffinPX) [reply src/lib/plugins/useItem.js:49]: Applying...

**Comment by rom1504** [comment]
Can you add a test so it doesn't get broken again in the future ?

**Review by rom1504 — APPROVED** [review]

**Pull request merged**